### ISSUE TITLE
static fields and accessors should be installed on the decorated class

### ISSUE THREAD

**Issue by JLHwung** [microsoft/TypeScript#57237]
### 🔎 Search Terms

decorator class static field 

### 🕗 Version & Regression Information

- This is a crash
- This changed between versions ______ and _______
- This changed in commit or PR _______
- [x] This is the behavior in every version I tried (after ES decorators are implemented), and I reviewed the FAQ for entries about decorators
- I was unable to test this on prior versions because _______

### ⏯ Playground Link

https://www.typescriptlang.org/play?useDefineForClassFields=true&target=9&ts=5.4.0-dev.20240130#code/DYUwLgBA8gTglgczgOwIbAGIHssG4BQ+AxsKgM5kQBCqMEA3gL6EBmArskWHFshACYgiACmxYAlA3wRo8JGkw4IAXghiCMmODYw+NGAWb4AAoKLFSFNUvrSIZMKm5EIABw33HziKiJEQFFh0qB4OTnAuALbCkkz4RkS8ZFigAHTAWAjC+qkAFuRQAO7IAAowWK4gMGAAnsIARK714gA01LR5BcVlFVW1DajNbTn5ZEWl5ZXVdfWRzeIEicjJaRlZsIgo6GKdY92TfTNNrbKbCjuj4z1T-fWDJxvy2zi7VwfTDXPiC0A

### 💻 Code

```ts
let OriginalFoo;

class Bar {}

function dec(Foo) {
  OriginalFoo = Foo;
  return Bar;
}

@dec
class Foo {
  static p;
  static accessor a;
  static m() {}
}

console.log(Bar.hasOwnProperty("p"), Bar.hasOwnProperty("a"), Bar.hasOwnProperty("m"));
console.log(OriginalFoo.hasOwnProperty("p"), OriginalFoo.hasOwnProperty("a"), OriginalFoo.hasOwnProperty("m"));
```

### 🙁 Actual behavior

It prints
```
false,  false,  false 
true,  true,  true 
```

### 🙂 Expected behavior

It should print
```
true, true, false
false, false, true
```

### Additional information about the issue

Per [RS: ClassDefinitionEvaluation](https://arai-a.github.io/ecma262-compare/?pr=2417&id=sec-runtime-semantics-classdefinitionevaluation): 
In step 27 the static methods are installed to the original class `originalFoo`.
In step 37 the receiver `F` is assigned by the decorated class `Bar`.
In step 40 the static fields and accessors are initialized on the decorated class `Bar`.
So `originalFoo` should contain all the static methods while `Bar` should contain all the static fields and accessors.

**Comment by JLHwung** [comment]
I checked the spec again and IIRC the static accessors should be installed on `OriginalFoo`, however, its backing storage private field should be installed on `Bar`, so accessing `OriginalFoo.a` should throw an error as the storage key is not defined on `OriginalFoo`, but `OriginalFoo.hasOwnProperty("a")` should return `true` anyway.

**Comment by rbuckton** [comment]
> It should print
> 
> ```
> true, true, false
> false, false, true
> ```

Per the spec (and as is reflected in your other comment), it should actually print
```
true, false, false
false, true, true
```


> accessing `OriginalFoo.a` should throw an error as the storage key is not defined on `OriginalFoo`

Yes, we currently emit `__classPrivateFieldGet(_classThis, _classThis, "f", _Foo_a_accessor_storage)` for the accessor when we should emit `__classPrivateFieldGet(this, _classThis, "f", _Foo_a_accessor_storage)` instead.

**Comment by JLHwung** [comment]
> Per the spec (and as is reflected in your other comment), it should actually print
> ```
> true, false, false
> false, true, true
> ```
@rbuckton Yes you are right, back then I confused the backing storage key definition with the accessor definition.
